src/ComponentBase.ts

```diff
@@ -325,10 +325,11 @@ export abstract class ComponentBase<P extends React.Props<any>, S extends Object
         }
     }
 
-    private _handleAutoSubscribe(store: StoreBase, key: string) {
+    private _handleAutoSubscribe(store: StoreBase, key: string): void {
         // Check for an existing auto-subscription.
-        if (this._hasMatchingAutoSubscription(store, key)) {
-            return;
+        const autoSubscription = this._findHandledAutoSubscription(store, key);
+        if (autoSubscription) {
+            return this._markAutoSubscriptionAsUsed(autoSubscription);
```

Comment:
Returning the `void` from `_markAutoSubscriptionAsUsed` is awkward. Put the `return` on its own.